CMakeLists.txt

```diff
@@ -225,23 +247,12 @@ if (ARCH_IA32 OR ARCH_X86_64)
     endif()
 endif()
 
-if (ARCH_AARCH64)
-    if (BUILD_SVE2_BITPERM)
-        set(GNUCC_ARCH "${GNUCC_ARCH}+sve2-bitperm")
```

Comment:
this is actually a more proper way to to it, the reason it fails right now is because it adds sve even if it was already in the GNUCC_ARCH variable.

Reply:
the end of autodetect flags for arch was `+nodotprod+sve`, by adding something after it, gcc would error out saying
```
Assembler messages:
Error: must specify extensions to add before specifying those to remove
```

possibly autodetect of tune/arch flags should be reconsidered in total?

Reply:
I agree that it should be reworked, but this is a low priority right now. , For that matter, I would also like to change build system altogether, but this is even lower priority. At the moment we have to adapt to the existing system. So, the goal is to detect is SVE* is available, but only actually enable it if BUILD_SVE* variables are set to true. I wonder if we should also allow building SVE on non-SVE systems -eg. a distro builder that lacks SVE, should be able to build the binaries.

Reply:
Interesting idea.
I believe it deserves its own attention.